Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Ride.hs

```diff
@@ -537,20 +538,92 @@ totalEarningsByFleetOwnerPerDriver fleetId driverId = do
-          filteredRide <- B.filter_'
-            ( \ride'' -> BeamR.driverId ride'' B.==?. B.val_ driverId.getId
+          rideDetails' <- B.filter_' (\rideDetails'' -> BeamRD.fleetOwnerId rideDetails'' B.==?. B.val_ fleetId) (B.all_ (SBC.rideDetails SBC.atlasDB))
+          B.filter_'
+            ( \ride'' -> ride''.status B.==?. B.val_ DDR.COMPLETED
+            )
+            do
+              B.join_' (SBC.ride SBC.atlasDB) (\ride'' -> BeamR.id ride'' B.==?. BeamRD.id rideDetails')
+  resTable <- case res of
+    Right res' -> mapM fromTType' res'
+    Left _ -> pure []
+  pure $ length resTable
+
+totalRidesByFleetOwnerPerVehicle :: MonadFlow m => Maybe Text -> Text -> m Int
+totalRidesByFleetOwnerPerVehicle fleetId vehicleNumber = do
+  dbConf <- getMasterBeamConfig
+  res <- L.runDB dbConf $
+    L.findRows $
+      B.select $
+        do
+          rideDetails' <- B.filter_' (\rideDetails'' -> (BeamRD.fleetOwnerId rideDetails'' B.==?. B.val_ fleetId) B.&&?. (BeamRD.vehicleNumber rideDetails'' B.==?. B.val_ vehicleNumber)) (B.all_ (SBC.rideDetails SBC.atlasDB))
+          B.filter_'
+            ( \ride'' -> ride''.status B.==?. B.val_ DDR.COMPLETED
+            )
+            do
+              B.join_' (SBC.ride SBC.atlasDB) (\ride'' -> BeamR.id ride'' B.==?. BeamRD.id rideDetails')
+  resTable <- case res of
+    Right res' -> mapM fromTType' res'
+    Left _ -> pure []
+  pure $ length resTable
+
+totalRidesByFleetOwnerPerDriver :: MonadFlow m => Maybe Text -> Id Person -> m Int
+totalRidesByFleetOwnerPerDriver fleetId driverId = do
+  dbConf <- getMasterBeamConfig
+  res <- L.runDB dbConf $
+    L.findRows $
+      B.select $
+        do
+          rideDetails' <- B.filter_' (\rideDetails'' -> BeamRD.fleetOwnerId rideDetails'' B.==?. B.val_ fleetId) (B.all_ (SBC.rideDetails SBC.atlasDB))
+          B.filter_'
+            ( \ride'' -> (ride''.status B.==?. B.val_ DDR.COMPLETED) B.&&?. (BeamR.driverId ride'' B.==?. B.val_ driverId.getId)
             )
             do
               B.join_' (SBC.ride SBC.atlasDB) (\ride'' -> BeamR.id ride'' B.==?. BeamRD.id rideDetails')
-          pure filteredRide
+  resTable <- case res of
+    Right res' -> mapM fromTType' res'
+    Left _ -> pure []
+  pure $ length resTable
+
+totalRidesByFleetOwnerPerVehicleAndDriver :: MonadFlow m => Maybe Text -> Text -> Id Person -> m Int
+totalRidesByFleetOwnerPerVehicleAndDriver fleetId vehicleNumber driverId = do
+  dbConf <- getMasterBeamConfig
+  res <- L.runDB dbConf $
+    L.findRows $
+      B.select $
+        do
+          rideDetails' <- B.filter_' (\rideDetails'' -> (BeamRD.fleetOwnerId rideDetails'' B.==?. B.val_ fleetId) B.&&?. (BeamRD.vehicleNumber rideDetails'' B.==?. B.val_ vehicleNumber)) (B.all_ (SBC.rideDetails SBC.atlasDB))
```

Comment:
why are we using multiple filter here ? Check findAllDriversWithInfoAndVehicle from person table for reference